packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -694,6 +697,53 @@ class _FloatingActionButtonTransitionState extends State<_FloatingActionButtonTr
 /// ```
 /// {@end-tool}
 ///
+/// ## Scaffold layout, the keyboard, and display "notches"
+///
+/// The scaffold will expand to fill the available space. That usually
+/// means that it will occupy its entire window or device screen. When
+/// the device's keyboard appears the Scaffold's ancestor [MediaQuery]
+/// widget's [MediaQueryData.viewInsets] changes and the Scaffold will
+/// be rebuilt. By default the scaffold's [body] is resized to make
+/// room for the keyboard. To prevent the resize set
+/// [resizeToAvoidBottomInset] to false. In either case the focused
+/// widget will be scrolled into view if it's within a scrollable
+/// container.
+///
+/// The [MediaQueryData.padding] value defines areas that might
+/// not be completely visible, like the display "notch" on the iPhone
+/// X. The scaffold's [body] is not inset by this padding value
+/// although an [appBar] or [bottomNavigationBar] will typically
+/// cause the body to avoid the vertical padding. The [SafeArea]
```

Comment:
I wouldn't over-emphasize 'vertical'. If you do call it out, add equal importance to horizontal paddings etc.

Reply:
In this case the comment is referring to the fact that when the appbar is visible, the body typically ends up below the bottom padding (assuming that the appbar has its default height). Similarly the bottom navbar typically causes the body to move out of the way of the bottom padding. Those are vertical adjustments.

Reply:
If you rotate the phone, the body just gets the insets horizontally (which is an equally important concept users should be aware of)

Reply:
Good point, the notches are horizontal in that case!